roles/cifmw_cephadm/tasks/pools.yml

```diff
@@ -47,19 +47,21 @@
 
 - name: Configure the RBD trash purge scheduler
   when:
-    - cifmw_enabled_services | default([]) | intersect(['cinder_volume'])
     - cifmw_cephadm_enable_trash_scheduler | default(false)
+    - cifmw_cephadm_pools is defined
+    - cifmw_cephadm_pools | length > 0
   block:
     - name: Get the RBD ceph_cli
       ansible.builtin.include_tasks: ceph_cli.yml
       vars:
         ceph_command: rbd
 
     - name: Set trash interval
+      when: item.application == 'rbd'
```

Comment:
This seems fragile to me, because if by any chance `application` is not set in the data structure, this task fails w/ `application` not defined.
am I right? @katarimanojk not sure you can double check this.

Reply:
I think most of the time people will use the hard coded `cifmw_cephadm_pools` and I think @katarimanojk was trying to set it for all RBD and exclude it for cephfs. However, I agree with @fmount that only want to set it for cinder, which will be called volumes. Unless someone overrides it, but in that case we'll have no way knowing which is the cinder volume unless we have some other sort of label. We could introduce a separate variable like `volumes_with_trash_purge` and default it to `['volumes']`. That at least makes a clear interface and allows the user to override which volumes will have it set. I'm open to other methods.

https://github.com/openstack-k8s-operators/ci-framework/blob/main/playbooks/ceph.yml#L273-L297

Reply:
Yeah, that solution is feasible but more complicated (I'm thinking for long term maintenance).
I was thinking more about something like "tagging" the pool where we want to enable trashing:

```
cifmw_cephadm_pools:
    name: volumes
    pg_autoscale_mode: True
    target_size_ratio: 0.3
    application: rbd
    trash_enabled: True
```

so we can keep iterating over the data structure, and enable the trash purge scheduler only for pools where we passed that `trash_enabled` boolean (that can be `|default(false)` for a general use case).
This should be simple enough and make that code less error prone.

Reply:
I like your proposal of the tag more. Let's go for that.

Reply:
@fmount Thanks for your suggestion, i will update the code accordingly.

Reply:
@fmount i have implemented your suggestion and tested it.